src/EnergyPlus/DaylightingManager.cc

```diff
@@ -10495,151 +10330,192 @@ void DayltgSetupAdjZoneListsAndPointers(EnergyPlusData &state)
     }
 
-    ZoneExtWin.deallocate();
+    enclExtWin.deallocate();
 }
 
-void CreateShadeDeploymentOrder(EnergyPlusData &state, int &ZoneNum)
+void CreateShadeDeploymentOrder(EnergyPlusData &state, int const enclNum)
 {
     // J. Glazer - 2018
     // create sorted list for shade deployment order
     // first step is to create a sortable list of WindowShadingControl objects by sequence
     std::vector<std::pair<int, int>> shadeControlSequence; // sequence, WindowShadingControl
     for (int iShadeCtrl = 1; iShadeCtrl <= state.dataSurface->TotWinShadingControl; ++iShadeCtrl) {
-        if (state.dataSurface->WindowShadingControl(iShadeCtrl).ZoneIndex == ZoneNum) {
-            shadeControlSequence.push_back(std::make_pair(state.dataSurface->WindowShadingControl(iShadeCtrl).SequenceNumber, iShadeCtrl));
+        for (int spaceNum : state.dataHeatBal->Zone(state.dataSurface->WindowShadingControl(iShadeCtrl).ZoneIndex).spaceIndexes) {
+            int shadeCtrlEnclNum = state.dataHeatBal->space(spaceNum).solarEnclosureNum;
+            if (shadeCtrlEnclNum == enclNum) {
+                shadeControlSequence.push_back(std::make_pair(state.dataSurface->WindowShadingControl(iShadeCtrl).SequenceNumber, iShadeCtrl));
+                break;
+            }
         }
     }
     // sort the WindowShadingControl objects based on sequence number
     sort(shadeControlSequence.begin(), shadeControlSequence.end());
     // now make the deployment list of lists.
     // each sublist is a group of surfaces that should be deployed together
     // often the sublist is just a single item.
-    for (auto sequence : shadeControlSequence) {
-        int curShadeControl = sequence.second;
-        if (state.dataSurface->WindowShadingControl(curShadeControl).MultiSurfaceCtrlIsGroup) {
-            // add a group of surfaces since they should be deployed as a group
-            std::vector<int> group;
-            for (int i = 1; i <= state.dataSurface->WindowShadingControl(curShadeControl).FenestrationCount; i++) {
-                group.push_back(state.dataSurface->WindowShadingControl(curShadeControl).FenestrationIndex(i));
-            }
-            state.dataDaylightingData->ZoneDaylight(ZoneNum).ShadeDeployOrderExtWins.push_back(group);
-        } else {
-            // add each individial surface as a separate list so they are deployed individually
-            for (int i = 1; i <= state.dataSurface->WindowShadingControl(curShadeControl).FenestrationCount; i++) {
-                std::vector<int> singleMemberVector;
-                singleMemberVector.push_back(state.dataSurface->WindowShadingControl(curShadeControl).FenestrationIndex(i));
-                state.dataDaylightingData->ZoneDaylight(ZoneNum).ShadeDeployOrderExtWins.push_back(singleMemberVector);
+    for (int controlNum : state.dataDaylightingData->enclDaylight(enclNum).daylightControlIndexes) {
+        auto &thisDaylightCtrl = state.dataDaylightingData->daylightControl(controlNum);
```

Comment:
thisDaylightCtrl is used twice, once in each of the if and else if. Does that save performance, or increase it?

Reply:
Yeah, that was just kind of a standard pattern, probably not useful for a small loop.